tests/e2e/tidbcluster/serial.go

```diff
@@ -1008,4 +1010,147 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 			klog.Info("success to check auto scale-in tidb to 2 replicas")
 		})
 	})
+
+	ginkgo.Context("[Verify: Upgrading Operator from 1.0.6", func() {
+		var oa tests.OperatorActions
+		var ocfg *tests.OperatorConfig
+		var version string
+
+		ginkgo.BeforeEach(func() {
+			version = os.Getenv("RELEASED_VERSION")
+			if len(version) < 1 {
+				version = "v1.0.6"
+			}
+			ocfg = &tests.OperatorConfig{
+				Namespace:   ns,
+				ReleaseName: "operator",
+				Tag:         version,
+				Image:       fmt.Sprintf("pingcap/tidb-operator:%s", version),
+			}
+			oa = tests.NewOperatorActions(cli, c, asCli, aggrCli, apiExtCli, tests.DefaultPollInterval, ocfg, e2econfig.TestConfig, nil, fw, f)
+			ginkgo.By("Installing CRDs")
+			oa.CleanCRDOrDie()
+			tests.DeployReleasedCRDOrDie(version)
+			ginkgo.By("Installing tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			oa.DeployOperatorOrDie(ocfg)
+		})
+
+		ginkgo.AfterEach(func() {
+			ginkgo.By("Uninstall tidb-operator")
+			oa.CleanOperatorOrDie(ocfg)
+			ginkgo.By("Uninstalling CRDs")
+			tests.CleanReleasedCRDOrDie(version)
+		})
+
+		ginkgo.It("Deploy TidbCluster and Upgrade Operator", func() {
+			tcName := "tidbcluster"
+			cluster := newTidbClusterConfig(e2econfig.TestConfig, ns, tcName, "", "")
+			cluster.Resources["pd.replicas"] = "3"
+			cluster.Resources["tikv.replicas"] = "3"
+			cluster.Resources["tidb.replicas"] = "2"
+			cluster.Monitor = false
+			cluster.OperatorTag = version
+			oa.DeployTidbClusterOrDie(&cluster)
+			oa.CheckTidbClusterStatusOrDie(&cluster)
+
+			getPodUids := func(ls string) ([]string, error) {
+				listOptions := metav1.ListOptions{
+					LabelSelector: ls,
+				}
+				podList, err := c.CoreV1().Pods(ns).List(listOptions)
+				if err != nil {
+					return nil, err
+				}
+				var uids []string
+				for _, pod := range podList.Items {
+					uids = append(uids, string(pod.GetUID()))
+				}
+				sort.Strings(uids)
+				return uids, nil
+			}
+
+			tc, err := cli.PingcapV1alpha1().TidbClusters(ns).Get(tcName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to get tc")
+
+			pdUids, err := getPodUids(labels.SelectorFromSet(label.New().Instance(tcName).PD().Labels()).String())
+			if err != nil {
+				framework.ExpectNoError(err, "failed to get pd pods uids")
+			}
+			tikvUids, err := getPodUids(labels.SelectorFromSet(label.New().Instance(tcName).TiKV().Labels()).String())
+			if err != nil {
+				framework.ExpectNoError(err, "failed to get tikv pods uids")
+			}
+			tidbUids, err := getPodUids(labels.SelectorFromSet(label.New().Instance(tcName).TiDB().Labels()).String())
+			if err != nil {
+				framework.ExpectNoError(err, "failed to get tidb pods uids")
+			}
```

Comment:
you can cache the pods before the upgrade and use [utilpod.PodsAreChanged](https://github.com/pingcap/tidb-operator/blob/716d8e393944f958365c1cb72c49a2bc82a460f5/tests/e2e/util/pod/pod.go#L30) to check if they are changed or not

https://github.com/pingcap/tidb-operator/blob/716d8e393944f958365c1cb72c49a2bc82a460f5/tests/e2e/tidbcluster/stability.go#L165-L170

check pod by name is good to know which pod changed or not changed

Reply:
updated.